pkg/controller/atlasproject/project.go

```diff
@@ -18,23 +19,24 @@ func ensureProjectExists(ctx *workflow.Context, connection atlas.Connection, pro
 	// Try to find the project
 	p, _, err := client.Projects.GetOneProjectByName(context.Background(), project.Spec.Name)
 	if err != nil {
-		return "", workflow.Terminate(workflow.ProjectNotCreatedInAtlas, err.Error())
-	}
-	if p.ID != "" {
-		ctx.Log.Debugw("Found Atlas Project", "id", p.ID)
-		return p.ID, workflow.OK()
-	}
-
-	// Otherwise try to create it
-	p = &mongodbatlas.Project{
-		OrgID: connection.OrgID,
-		Name:  project.Spec.Name,
+		var apiError *mongodbatlas.ErrorResponse
+		if errors.As(err, &apiError) && apiError.ErrorCode == atlas.NotInGroup {
+			// Project doesn't exist? Try to create it
+			p = &mongodbatlas.Project{
+				OrgID: connection.OrgID,
+				Name:  project.Spec.Name,
+			}
+			if p, _, err = client.Projects.Create(context.Background(), p); err != nil {
+				return "", workflow.Terminate(workflow.ProjectNotCreatedInAtlas, err.Error())
+			}
+			ctx.Log.Infow("Created Atlas Project", "name", project.Spec.Name, "id", p.ID)
+		} else {
+			return "", workflow.Terminate(workflow.ProjectNotCreatedInAtlas, err.Error())
+		}
 	}
-
-	if p, _, err = client.Projects.Create(context.Background(), p); err != nil {
-		return "", workflow.Terminate(workflow.ProjectNotCreatedInAtlas, err.Error())
+	if p == nil || p.ID == "" {
+		log.Error("Project or its project ID are empty")
```

Comment:
Please change to `ctx.Log` - or maybe we could use this error msg as a `Reason`?

Reply:
ah, that was an auto-completion so the `"github.com/prometheus/common/log"` is used here.. 🤦 

good catch